File: examples/consumer.py
import json, os
from pathlib import Path
import time
from urllib.request import urlopen
from laofu_browser import BrowserClient
c = BrowserClient(os.environ['LAOFU_URL'], os.environ['LAOFU_TOKEN'])
profile_id, source = os.environ['LAOFU_PROFILE'], os.environ['LAOFU_SAMPLE_URL']
Path('upload.bin').write_bytes(bytes([43])*(12*1024*1024))
upload = c.upload('upload.bin'); c.download(upload['id'], 'uploaded-copy.bin')
task = c.submit_task({'type':'article.capture@v1','execution':{'profileId':profile_id},'input':{'url':source+'/article'}}, 'py-capture')
done = c.wait(task['id']); assert done['state'] == 'succeeded', done
c.download(next(a['id'] for a in done['artifacts'] if a['filename'].endswith('.zip')), 'article.zip')
gate = c.submit_task({'type':'article.capture@v1','execution':{'profileId':profile_id},'input':{'url':source+'/gate?client=py'}}, 'py-gate')
assert c.wait(gate['id'])['state'] == 'waiting_user'
with urlopen(source+'/unlock?client=py') as response: response.read()
time.sleep(0.8); c.resume(gate['id']); resumed = c.wait(gate['id']); assert resumed['state'] == 'succeeded', resumed
cancelled = c.submit_task({'type':'article.capture@v1','execution':{'profileId':os.environ['LAOFU_OFFLINE_PROFILE']},'input':{'url':source+'/article'}}, 'py-cancel')
assert c.cancel(cancelled['id'])['state'] == 'cancelled'
Path('result.json').write_text(json.dumps({'sdk':'python','capture':done['id'],'resumed':resumed['id'],'cancelled':cancelled['id'],'upload':upload['id'],'state':done['state']}))
print('Python installed package: capture/upload/download/query/cancel/resume passed')

File: examples/consumer.mjs
import fs from "node:fs";
import path from "node:path";
import { BrowserClient } from "@laofu/browser";
const c = new BrowserClient(process.env.LAOFU_URL, process.env.LAOFU_TOKEN),
  profileId = process.env.LAOFU_PROFILE;
const source = process.env.LAOFU_SAMPLE_URL;
const file = "upload.bin";
fs.writeFileSync(file, Buffer.alloc(12 * 1024 ** 2, 42));
const uploaded = await c.upload(file);
await c.download(uploaded.id, "uploaded-copy.bin");
const capture = await c.submitTask(
  {
    type: "article.capture@v1",
    execution: { profileId },
    input: { url: source + "/article" },
  },
  "ts-capture",
);
const done = await c.wait(capture.id);
if (done.state !== "succeeded") throw new Error(JSON.stringify(done.error));
await c.download(
  done.artifacts.find((a) => a.filename.endsWith(".zip")).id,
  "article.zip",
);
const gate = await c.submitTask(
  {
    type: "article.capture@v1",
    execution: { profileId },
    input: { url: source + "/gate?client=ts" },
  },
  "ts-gate",
);
const waiting = await c.wait(gate.id);
if (waiting.state !== "waiting_user") throw new Error("expected waiting_user");
await fetch(source + "/unlock?client=ts");
await new Promise((r) => setTimeout(r, 800));
await c.resume(gate.id);
const resumed = await c.wait(gate.id);
if (resumed.state !== "succeeded") throw new Error("resume did not finish");
const cancelled = await c.submitTask(
  {
    type: "article.capture@v1",
    execution: { profileId: process.env.LAOFU_OFFLINE_PROFILE },
    input: { url: source + "/article" },
  },
  "ts-cancel",
);
if ((await c.cancel(cancelled.id)).state !== "cancelled")
  throw new Error("queued cancel failed");
fs.writeFileSync(
  "result.json",
  JSON.stringify({
    sdk: "typescript",
    capture: done.id,
    resumed: resumed.id,
    cancelled: cancelled.id,
    upload: uploaded.id,
    state: done.state,
  }),
);
console.log(
  "TS installed package: capture/upload/download/query/cancel/resume passed",
);
